syntax/peek.vim

```diff
@@ -3,15 +3,19 @@ if exists("b:current_syntax")
 endif
 
 for s:section in ['Numerical', 'Literal', 'Special']
-	execute 'syntax match Reg' . s:section . ' /^' . s:section . '.*/'
+	execute 'syntax match RegName' . s:section . ' /^' . s:section . '/ contained'
+	execute 'syntax match Reg' . s:section . ' /^' . s:section . '.*/ contains=RegName' . s:section
+	execute 'highlight default link RegName' . s:section . ' Title'
 	execute 'highlight default link Reg' . s:section . ' Label'
 endfor
 unlet s:section
 
-syntax match NumReg /^[0-9]:/
-syntax match LitReg /^[A-Za-z]:/
+syntax match RegTitle /^\s*Registers/
```

Comment:
I believe here we should not hardcode the title, as this is customisable. We should rather refer to the value that is contained in the dictionary instead, which means something along the lines of
```
let s:window_title = luaeval(require('nvim-peekup.config').geometry["title"])
syntax match RegTitle /^\s*/ . s:window_title . /
```
where my escaping is clearly wrong here. [luaeval](https://github.com/nanotee/nvim-lua-guide#luaeval) evaluates a Lua expression string and returns its value, so this should work (up to the correct grammar).

Reply:
Fair point. See updated below

Reply:
It works just fine!